src/Workspaces/Core/Portable/PublicAPI.Unshipped.txt

```diff
@@ -3,6 +3,10 @@ Microsoft.CodeAnalysis.Editing.DeclarationModifiers.IsVolatile.get -> bool
 Microsoft.CodeAnalysis.Editing.DeclarationModifiers.WithIsExtern(bool isExtern) -> Microsoft.CodeAnalysis.Editing.DeclarationModifiers
 Microsoft.CodeAnalysis.Editing.DeclarationModifiers.WithIsVolatile(bool isVolatile) -> Microsoft.CodeAnalysis.Editing.DeclarationModifiers
 Microsoft.CodeAnalysis.Editing.SyntaxGenerator.ElementBindingExpression(params Microsoft.CodeAnalysis.SyntaxNode[] arguments) -> Microsoft.CodeAnalysis.SyntaxNode
+Microsoft.CodeAnalysis.Options.OptionSet.GetOption(Microsoft.CodeAnalysis.Options.OptionKey optionKey) -> object
+*REMOVED*override Microsoft.CodeAnalysis.Options.DocumentOptionSet.GetOption(Microsoft.CodeAnalysis.Options.OptionKey optionKey) -> object
+*REMOVED*abstract Microsoft.CodeAnalysis.Options.OptionSet.GetOption(Microsoft.CodeAnalysis.Options.OptionKey optionKey) -> object
```

Comment:
NOTE: Even though theoretically this change introduces a breaking change for all public sub-types of OptionSet by making an abstract public method a non-abstract method, it is not really a breaking change as OptionSet already has internal abstract methods, hence cannot be sub-typed outside Roslyn. I also verified none of our IVT partners sub-type OptionSet.

https://github.com/dotnet/roslyn/blob/75fb12b5cb179a244830647729449e04eab47834/src/Workspaces/Core/Portable/Options/OptionSet.cs#L104

Reply:
Another possible option here is to revert the abstract to non-abstract breaking API change, and call out `(T)GetOption(optionKey)` as an unsupported/deprecated operation in favor of invoking `GetOption<T>(optionKey)` which provides type safe access to option value. Former was relying on internal implementation details, which were never part of the Options API contract.

Reply:
Restricting overrides of this method is not a breaking change for this type, so if the approach works then the approach works.

Reply:
Agreed no concern here since this was never practically inheritable.